pandas/core/window/rolling.py

```diff
@@ -490,6 +491,9 @@ def _apply_blockwise(
             return self._apply_series(homogeneous_func)
 
         obj = self._create_data(self._selected_obj)
+        if self.axis == 1:
+            obj = obj.astype(infer_dtype_from(obj.values)[0], copy=False)
```

Comment:
This logic may be better suited in `_create_data` so other operations also benefit when axis=1

Reply:
Also generally we require all input data to be castable to float (see the `_prep_values` function).

So instead of using `infer_dtype_from` I think you can just try to cast directly to float instead.

Reply:
I found an issue discussing adding support for other dtypes. I wanted to be on the save side with regard to this. But I change it then.

Additionally I was not sure if this should affect functions outside of ``_apply_blockwise``. I moved this into ``_create_data``

Reply:
Ähm I got one question about the dtype thing: How can we convert to float32, if float32 is the input and float64 otherwise? Just giving "float" into astype converts it to "float64"

Reply:
Right eventually we would like to support more input data types eventually but that issue can be addressed later.

We generally coerce all inputs to `float64` in `_prep_values` anyways so the 32 vs 64 bit distinction doesn't matter here. So this should always coerce to float64

Reply:
Thanks very much. Done